aerogear/unifiedpush-operator
# Start working on Monitoring feature + Improvements based on the first contact with

## Motivation:
- Getting Started to use it like any other developer
- Prepare to start to impl monitor features
- https://issues.jboss.org/browse/INTLY-2619

NOTE: The Monitor changes are required for:
* https://issues.jboss.org/browse/INTLY-1441
* https://issues.jboss.org/browse/INTLY-1442
* https://issues.jboss.org/browse/INTLY-1444

## What
- Improvements into the README and the Getting Started process
- Improvements in the started monitor process and commands
- Add alerts, Prometheus rules and graphane dashboard for the operator. 

## How
- Improve the steps to Getting Started with
- Add pre-requirements
- Add missing sections in the README ( contribution, questions )
- Move content of limitations to Supportability 
- Create make install/uninstall to make easier the first contact with
- Organize makefile
- Create a table with the makefile commands to make easier to use it as to improve the Maintenace ability. 
- Add the github files
- Add monitor dir and move service_monitor.yaml for there 
- Add graphana dashboard and Prometheus rules and alerts for the operator. 

## Why
- Make easier the first contact
- Following the standards
- Improve maintenance ability
- Attend the requirements
- Prepare the base to start to add the UPS server monitor files. 

## Steps to verify:
- Following the getting started steps
- Check the monitor here: https://grafana-route-middleware-monitoring.apps.london-4bc4.openshiftworkshop.com/d/null/ups-operator?refresh=10s&orgId=1

## Checklist:


## Progress
